### ISSUE TITLE
Don't display the blue notice bar until a user selects either a GA4 or UA property when connecting Analytics. 

### ISSUE THREAD

**Issue by jamesozzie** [google/site-kit-wp#3614]
When setting up Analytics and selecting an account which has **both a UA and GA4 property** don't display the blue notice bar regarding the creation of an additional property until a user makes a selection. 

This would lead to less confusion considering the notice bar that appears on account selection assumes the user will select a UA property. 

![2021-06-22-14-03-27](https://user-images.githubusercontent.com/41326532/122929632-14da3580-d363-11eb-8de5-4be288fb6bac.gif)

_Screenshot of the current blue bar that appears upon selecting an Analytics account with both a UA and GA4 property._ 

![image](https://user-images.githubusercontent.com/41326532/122930296-c4afa300-d363-11eb-8ece-17b493afbc9c.png)

The blue notice assumes a user will select a UA property. If the user selects a GA4 property the notice does change, which is ideal however an improvement would be to only display that notice when a property selection has been made. 

![image](https://user-images.githubusercontent.com/41326532/122930857-3ee02780-d364-11eb-8d19-2a489c00273c.png)

## Steps to reproduce

1. Setup Site Kit 
2. Connect Analytics
3. Select an account that has both a UA and GA4 property
4. The blue notice bar appears before the user makes a property selection, assuming they will select a UA property. 

## Additional Context

 - PHP Version: 
 - Plugin Version [e.g. 22]

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* In the Analytics setup flow (with `ga4Setup` flag active), any of the three new blue notices around the associated other property (whether about creating a new one or selecting a matching one) should only appear when a property is already selected in the primary dropdown.
    * In other words, if no account/property is selected or if only an account has been selected, the notice should not be visible _yet_.

## Implementation Brief

* Update the `SetupFormUA` component:
  * Pull UA propertyID using the `getPropertyID` selector
  * Update the `shouldShowGA4PropertyNotice` constant to also check if the propertyID is not empty.
* Update the `SetupFormGA4` component:
  * Pull GA4 propertyID using the `getPropertyID` selector
  * Update the `GA4PropertyNotice` element to render it only when the `propertyID` is not empty.
* Update the `SetupFormGA4Transitional` component:
  * Pull GA4 propertyID using the `getPropertyID` selector of the analytics-4 module
  * Update the `GA4PropertyNotice` element to render it only when either UA or GA4 propertyID is not empty.

### Test Coverage

* N/A

### Visual Regression Changes

* N/A

## QA Brief

* Setup Site Kit 
* Connect Analytics using an account with a bunch of properties but none that matches the current site URL
* Check that the blue notice bar for the GA4 property does not appear before the user makes a property selection

## Changelog entry

* Show information notice about associated UA / GA4 property only once a property has been selected.

**Comment by tofumatt** [comment]
IB ✅

**Comment by Hazlitte** [comment]
I have been working on the execution of this and I seem to be unable to test it in my local environment.  I suspect this is because I don't have permissions to create properties with my 10up account.

I built a release zip and uploaded it to a site I have which does have both UA and GA4 properties.   However I am struggling to recreate the scenario described above as the Analytics setup flow is automatically selecting the UA property in the store.  It does not display 'Property' for me in the drop down list and that option is not in the list when I click on it.

How do I get the 'Property' drop down list field to display 'Property' as the first option when connecting Analytics?

![Screenshot 2021-07-06 at 15 43 27](https://user-images.githubusercontent.com/5174233/124620209-2d714200-de71-11eb-9eaa-62156b56c434.png)

**Comment by tofumatt** [comment]
> I suspect this is because I don't have permissions to create properties with my 10up account.

You don't need to create a property here, but it looks like one is automatically being selected for you on said site because you have a matching property for it already.

You can test this by loading a site where you don't have a matching property and are forced to select one. Or by selecting a GA _Account_ that doesn't have a matching property for the site you're on. Here's a screencast:

![2021-07-06 16 38 05](https://user-images.githubusercontent.com/90871/124628778-b93a9c80-de78-11eb-85a7-71fca88e891b.gif)

**Comment by Hazlitte** [comment]
@tofumatt Thank you.  I have tested the code changes outlined in the IB on a site using an account that does not have a matching property.  In the `SetupFormGA4Transitional` component the GA4 property id is being returned as `property_create`.  So it is not empty and the blue message is still displayed.  For this component should the IB in fact say:

"Update the GA4PropertyNotice element to render it only when either UA or GA4 propertyID is not empty and GA4 propertyID does not contain `property_create`."

And should the conditional in the `SetupFormGA4` be updated in a similar way?

**Comment by tofumatt** [comment]
Yes, I think in fact there's never an "empty" value but rather a magic constant value like `'property_create'` is used when the selection is empty. See: https://github.com/google/site-kit-wp/blob/93d2c1b3e902cd4d91aadf82e029d0d5fabaa830/assets/js/modules/analytics/components/common/PropertySelect.js#L89-L109

So yes, if a property isn't selected (eg the _selection_ is empty and the select's _value_ is the `PROPERTY_CREATE` constant value), then don't display the notice. If a value is selected, display the correct notice 👍🏻

**Comment by Hazlitte** [comment]
Thanks @felixarntz

**Comment by tofumatt** [comment]
@felixarntz Oh shoot, thanks, yes. My bad, not sure why I spaced on that one 😅

**Comment by felixarntz** [comment]
@wpdarren Good catch!

I guess the logic specifically for the `ga4transitional` case needs to be slightly adjusted: Instead of just checking whether one of the two properties is already specified (which here is the case since GA4 property is set to create a new one), this should only consider the value of whether the _primary_ property is already set. In other words, we also need to use `getPrimaryPropertyType`.

@eugene-manuilov Could you coordinate getting a follow-up PR ready for this against `main` that adjusts this one case?

**Comment by eugene-manuilov** [comment]
@felixarntz do you mind reviewing my PR #3699? It contains bug fixes that we discussed on our call today.

**Comment by wpdarren** [comment]
### QA Update: Pass ✅ 

- Verified that within Analytics set up flow, that when you select an account with lots of properties not linked with the URL of the test site, that the blue notice no longer appears. 
- Verified that only when you select a property does the appropriate blue notice appear. 

![image](https://user-images.githubusercontent.com/73545194/125773185-73c43968-5258-46e9-b17f-517b5f831fb3.png)
